WooCommerce/Classes/Authentication/WPComLogin/WPComEmailLoginViewModel.swift

```diff
@@ -78,8 +87,16 @@ final class WPComEmailLoginViewModel: ObservableObject {
             }
             await startAuthentication(email: email, isPasswordlessAccount: passwordless)
         } catch {
-            analytics.track(event: .JetpackSetup.loginFlow(step: .emailAddress, failure: error))
-            onError(error.localizedDescription)
+            if allowAccountCreation,
+               let apiError = error as? WordPressAPIError<WordPressComRestApiEndpointError>,
+               case let .endpointError(endpointError) = apiError,
+               endpointError.apiErrorCode == Constants.unknownUserErrorCode {
+                // The user does not exist yet, trigger magic link flow for account creation
+                await requestAuthenticationLink(email: email, forAccountCreation: true)
```

Comment:
Not specifically related to functionality of this PR: I feel it might be valuable to add tracking for this case. What do you think?

Reply:
Yes, it's already planned, I'll add some tracking before releasing the feature.